### ISSUE TITLE
[Mapping] Make CobolVisitor Apply also the Syntax Analysis

### ISSUE THREAD

**Issue by temanbrcom** [eclipse/che-che4z-lsp-for-cobol#339]
## Description ##
As a SE, I would like CobolVisitor to apply both syntax and semantic analysis so I can remove extra steps of the analysis.

## Acceptance Criteria ##
- Remove the parser syntax listener and delegate its work to the visitor
- Syntax analysis remain the same
- The positions for the errors and semantic elements retrieved correctly

## Definition of Done ##
- The story is implemented according to the description.
- Unit testing done (Preferably by TDD approach, coverage should be at least 70%).
- Code review done by at least one other team member.
- Manual testing is done.
- Documentation is added and reviewed (if needed).
- The pull request is approved.
- The story is merged into the development branch.
- Tests are performed on the development branch.

**Comment by temanbrcom** [comment]
Blocked by #220

**Comment by temanbrcom** [comment]
This issue cannot be implemented without losing some syntax checks. Better implement the same logic with the late mapping of positions, passing all the checks from Lexer to Parser
